api/src/main/java/marquez/api/JobResource.java

```diff
@@ -273,6 +274,47 @@ public Response getRunFacets(
     return Response.ok(facets).build();
   }
 
+  @Timed
+  @ResponseMetered
+  @ExceptionMetered
+  @POST
+  @Path("/namespaces/{namespace}/jobs/{job}/tags/{tag}")
+  @Produces(APPLICATION_JSON)
+  public Response updatetag(
+      @PathParam("namespace") NamespaceName namespaceName,
+      @PathParam("job") JobName jobName,
+      @PathParam("tag") TagName tagName) {
+    throwIfNotExists(namespaceName);
+    throwIfNotExists(namespaceName, jobName);
+
+    jobService.updateJobTags(namespaceName.getValue(), jobName.getValue(), tagName.getValue());
+    Job job =
+        jobService
+            .findJobByName(namespaceName.getValue(), jobName.getValue())
+            .orElseThrow(() -> new JobNotFoundException(jobName));
```

Comment:
Yeah understand what you mean -it will fall over before it gets to this point so why bother? Mainly as that seems to be the de-facto pattern for a lot of the code i.e

execute something -> retrieve object else throw an error.

Reply:
Thanks for following our code style!